libdrgn/program.c

```diff
@@ -635,8 +635,10 @@ static void drgn_program_set_language_from_main(struct drgn_program *prog,
 	struct drgn_dwarf_index_iterator it;
 	static const uint64_t tags[] = { DW_TAG_subprogram };
 
-	drgn_dwarf_index_iterator_init(&it, dindex, "main", strlen("main"),
-				       tags, ARRAY_SIZE(tags));
+	err = drgn_dwarf_index_iterator_init(&it, &dindex->ns, "main", strlen("main"),
+					     tags, ARRAY_SIZE(tags), true);
+	if (err)
+		drgn_error_destroy(err);
```

Comment:
I don't know how to handle this - just making sure this is ok

Reply:
This is fine, but we need to return, not march forward with an invalid iterator.